tests/oauth/test_base.py

```diff
@@ -69,97 +67,104 @@ def json(self):
-    query = check_oauth_redirect_url(
-        response=response, client=client, path='/admin/oauth/authorize', scope=TEST_DATA.app_scopes
-    )
-    state = check_oauth_redirect_query(query=query, scope=TEST_DATA.app_scopes)
-
-    # Callback calls to get tokens
-    shopify_request_mock = mocker.patch('httpx.AsyncClient.request', new_callable=AsyncMock)
-    shopify_request_mock.side_effect = [
-        MockHTTPResponse(status_code=200, jsondata=OFFLINETOKEN_DATA),
-        MockHTTPResponse(status_code=200, jsondata=ONLINETOKEN_DATA),
-    ]
-    # --------- Test the callback endpoint for installation -----------
-    query_str = urlencode(
-        dict(shop=TEST_STORE, state=state, timestamp=now_epoch(), code='INSTALLCODE')
-    )
-    hmac_arg = hmac.calculate_from_message(secret=SHOPIFY_SECRET_KEY, message=query_str)
-    query_str += '&hmac=' + hmac_arg
-
-    response = client.get('/callback', params=query_str, allow_redirects=False)
-    query = check_oauth_redirect_url(
-        response=response,
-        client=client,
-        path='/admin/oauth/authorize',
-        scope=TEST_DATA.user_scopes,
-    )
-    state = check_oauth_redirect_query(
-        query=query,
-        scope=TEST_DATA.user_scopes,
-        query_extra={'grant_options[]': ['per-user']},
-    )
-
-    assert await shopify_request_mock.called_with(
-        method='post',
-        url=f'https://{TEST_STORE}/admin/oauth/access_token',
-        json={
-            'client_id': SHOPIFY_API_KEY,
-            'client_secret': SHOPIFY_SECRET_KEY,
-            'code': 'INSTALLCODE',
-        },
-    )
-
-    TEST_DATA.post_install.assert_called_once()
-    TEST_DATA.post_install.assert_called_with('test', OfflineToken(**OFFLINETOKEN_DATA))
-
-    # --------- Test the callback endpoint for login -----------
-    query_str = urlencode(
-        dict(shop=TEST_STORE, state=state, timestamp=now_epoch(), code='LOGINCODE'), safe='=,&/[]:'
-    )
-    hmac_arg = hmac.calculate_from_message(secret=SHOPIFY_SECRET_KEY, message=query_str)
-    query_str += '&hmac=' + hmac_arg
-
-    response = client.get('/callback', params=query_str, allow_redirects=False)
-    state = check_oauth_redirect_url(
-        response=response,
-        client=client,
-        path=f'/admin/apps/{HANDLE}',
-        scope=TEST_DATA.user_scopes,
-    )
-
-    assert await shopify_request_mock.called_with(
-        method='post',
-        url=f'https://{TEST_STORE}/admin/oauth/access_token',
-        json={
-            'client_id': SHOPIFY_API_KEY,
-            'client_secret': SHOPIFY_SECRET_KEY,
-            'code': 'LOGINCODE',
-        },
-    )
-
-    TEST_DATA.post_login.assert_called_once()
-    TEST_DATA.post_login.assert_called_with('test', OnlineToken(**ONLINETOKEN_DATA))
+    with pytest.raises(FastAPIImportError):
+        from spylib.oauth import OfflineToken, OnlineToken, init_oauth_router
+
+        app = FastAPI()
```

Comment:
I think there is a more readable way to do this that avoids having `except ImportError: pass` which is a bit of a no-no, at least I don't want to give a precedent for it.

Also avoids the big change in indentation ... always nice for git diff :wink: 

```suggestion
    if 'fastapi' not in sys.modules and importlib.util.find_spec('fastapi') is None:
        with pytest.raises(FastAPIImportError):
            from spylib.oauth import OfflineToken, OnlineToken, init_oauth_router
        return

    from spylib.oauth import OfflineToken, OnlineToken, init_oauth_router
    
    app = FastAPI()
```